torch/autograd/_functions/blas.py

```diff
@@ -17,8 +17,24 @@ class Addmm(InplaceFunction):
 
     @staticmethod
     def symbolic(g, add_matrix, matrix1, matrix2, beta=1, alpha=1, inplace=False):
-        # TODO: test if broadcasting occurred
-        return g.op("Gemm", matrix1, matrix2, add_matrix, beta_f=beta, alpha_f=alpha, broadcast_i=True)
+        assert matrix1.hasType() and matrix2.hasType() and add_matrix.hasType()
+        sizes1 = matrix1.type().sizes()
+        sizes2 = matrix2.type().sizes()
+        sizes_add = add_matrix.type().sizes()
+        broadcast = False
+        not_supported = False
+        assert len(sizes1) == 2 and len(sizes2) == 2 and len(sizes_add) <= 2 and len(sizes_add) > 0
+        if len(sizes_add) == 1:
```

Comment:
At some point we should factor this logic out, because there are going to be a number of other places where we need to detect and support a broadcast (e.g., Add)